lib/nfg_ui/bootstrap/utilities/modalable.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+module NfgUi
+  module Bootstrap
+    module Utilities
+      attr_reader :remote
+      # Allows a component to utilize the :modal option
+      # Which then automatically formats the component's data attributes
+      # to connect to the desired modal. Note the modal option requires the '#'
+      # preceding the ID per bootstrap docs ('#the_modal_id' not 'the_modal_id')
+      # 
+      # Usage notes:
+      # 1. When to use:
+      # The :modal option should only be used when calling to a modal
+      # that has been embedded on the page (not for remote links, more below).
+      # There are no detrimental effects to using the modal option on a 
+      # component that does not have a corresponding modal, however it
+      # does add useless data attributes to your component.
+      # 
+      # 2. Regarding ajax remotely rendered modals
+      # When leveraging a :remote link that renders a modal remotely,
+      # the component does _not_ need the modal option. Your action's js.erb
+      # will/should render the partial within the modal as part of its javascript.
+      # 
+      # Example usage for (non-remote) embedded modal:
+      # The modalable component:
+      # ui.bootstrap(:button, body: 'Modal Button', modal: '#the_target_modal_id')
+      #
+      # The actual modal to be activated, embedded on the page:
+      # ui.bootstrap(:modal, body: 'The modal', id: 'the_target_modal_id')
+      module Modalable
+        def modal
+          options.fetch(:modal, nil)
+        end
+
+        # Only overwrite data-toggle and data-target
+        # The operating assumption is that activating a modal is more important
+        # and thus, will take precedence over less valuable competing data toggles
+        #
+        # For example: ui.nfg(:button, modal: '#the_target_modal', tooltip: 'The Tooltip')
+        # will ignore the tooltip (since tooltip is initialized via a competing data-toggle)
+        def data
+          return super if remote
```

Comment:
@hoenth do you think it'd be better / more useful to raise an `ArgumentError` instead of just setting the component to ignore it? My gut says just ignore it, there could be instances where it's necessary to have both (e.g.: a conditional :remote trait).

Reply:
Can you think of a real life instance where you would want it to be conditional? The problem with a silent failure is that you would have to dig down into the meat of the code to find out what is happening. I would suggest raising an error until we have a specific case where both attributes are needed

Reply:
Awesome point. I agree. I can’t think of a real world situation